tiledb/sm/fragment/fragment_metadata.cc

```diff
@@ -870,7 +872,9 @@ Status FragmentMetadata::write_footer(Buffer* buff) const {
   RETURN_NOT_OK(write_file_var_sizes(buff));
   RETURN_NOT_OK(write_file_validity_sizes(buff));
   RETURN_NOT_OK(write_generic_tile_offsets(buff));
-
+  if (version_ >= 10) {
```

Comment:
I saw version `8` somewhere else, let's make another quick test to see if we are using the versions correctly.

Reply:
Also let's make sure that this change does not violate backwards compatibility by adding older version arrays to the tests.

Reply:
We have backwards compatibility arrays already, and this change is currently breaking them, see the failures in the ["serialization" CI jobhttps://dev.azure.com/TileDB-Inc/CI/_build/results?buildId=11794&view=logs&j=394820b7-6fbf-58b1-3c3e-2916b88bc939&t=754239f3-192d-57c7-b335-f5140b471bfd&l=1172).

@bdeng-xt the changes needed for compatibility are as follows:
1) If the `__schema` folder does not exist, we need to fall back to the `__array_schema.tdb` file
2) If the `__schema` folder does exist BUT we have any fragment which is older than fragment version 10 we need to load the `__array_schema.tdb` file, and use this for the schema of those older fragments.

Reply:
will check fragment version for older fragments and load __array_schema.tdb

Reply:
Version 10 is correct, assuming this does not make it into the 2.3 release.